### ISSUE TITLE
PRs should ignore merge commits

### ISSUE THREAD

**Issue by TheFriendlyCoder** [go-gitea/gitea#30760]
### Feature Description

When a new PR gets created and the source branch matches the commit history of the target branch exactly with the exception of a merge commit, the web UI warns you the the source branch is out of date. While this is, strictly speaking, accurate, merge commits don’t actually contain any source code changes and so this warning can be ignored. However, if there are actual code commits on the target branch then it may not be safe to ignore this warning. Without doing a full comparison between the branches it is impossible to know for sure. This creates unnecessary confusion for users.

I am pretty sure other git repository tools like GitHub and Gitlab exclude merge commits when doing history comparisons to avoid this problem.

This issue is particularly problematic for developers that regularly stack commits across multiple PRs because every PR after the first will inherit this branch out of date warning

**Comment by AdamMajer** [comment]
> merge commits don’t actually contain any source code changes 

This is not true. You can have anything there. Merge commits are just commits with multiple parents, but otherwise, it's just a commit with its own tree.
